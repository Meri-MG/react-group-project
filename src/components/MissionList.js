import { useEffect } from 'react';
import { useSelector, useDispatch } from "react-redux";
import Missions from "./Mission";
import { getMissionFromApi } from '../redux/actions/rocket';

const MissionList = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionReducer);
  console.log(missions.missions, 'this me searching');
  useEffect(() => {
    dispatch(getMissionFromApi());
  }, []);
  return (
    <div className="title">
      {missions.missions.map((mission) => (
        <Missions
          key={mission.id}
          mission={mission}
        />
      ))}
    </div>
  );
};

export default MissionList;