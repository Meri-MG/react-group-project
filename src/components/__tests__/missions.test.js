import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import MissionList from '../missions/MissionList';
import MissionBtn from '../missions/missionbtn';

describe('Test  Mission section functionality ', () => {
  test('snapshot', () => {
    const tree = renderer.create(<Provider store={store}><MissionList /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test click ', async () => {
    const handleJoin = jest.fn();
    const join = { };
    render(<MissionBtn handleJoin={handleJoin} join={join} /> );
    await userEvent.click(screen.getByRole('button'));
    expect(handleJoin).toHaveBeenCalled();
  });
  test('test button text ', async () => {
    const handleJoin = jest.fn();
    const join = { };
    render(<MissionBtn handleJoin={handleJoin} join={join} /> );
    await userEvent.click(screen.getByRole('button'));
    expect(screen.getByRole('button').textContent).toEqual('Leave Mission');
  });
});
