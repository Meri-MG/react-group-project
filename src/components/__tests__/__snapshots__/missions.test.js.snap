// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test  Mission section functionality  snapshot 1`] = `
<div
  className="missions"
>
  <table
    className="miss-table"
  >
    <thead>
      <tr>
        <td
          className=" col-head"
        >
          Mission
        </td>
        <td
          className=" col-head"
        >
          Description
        </td>
        <td
          className=" col-head"
        >
          Status
        </td>
        <td
          className=" col-head"
        >
          Join
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
